release.md

```diff
@@ -0,0 +1,298 @@
+$ truffle compile --list prereleases       # Recent prereleases from solc-bin (JS)
+$ truffle compile --list docker            # Recent docker tags from hub.docker.com
+$ truffle compile --list releases --all    # Complete list of stable releases.
+```
+
+### Specify a solcjs version in `truffle.js`
+
+Set the compilers solc `version` key to the one you'd like. Truffle will fetch it from the solc-bin server and cache it in your local evironment.
+```javascript
+module.exports = {
+  networks: {
+    ... etc ...
+  },
+  compilers: {
+     solc: {
+       version: <string>  // ex:  "0.4.20". (Default: truffle's installed solc)
+     }
+  }
+};
+```
+
+### Advanced:
++ docker
++ native binary
++ `path/to/solc`
++ solc settings
+
+Docker images should be installed locally by running:
+```shell
+$ docker pull ethereum/solc:0.4.22  // Example image
+```
+**truffle.js**
+```javascript
+// Native binary
+compilers: {
+  solc: {
+    version: "native"
+  }
+}
+
+// Docker
+compilers: {
+  solc: {
+    version: "0.4.22",   // Any published image name
+    docker: true
+  }
+}
+
+// Relative or absolute path to an npm installed solc-js
+compilers: {
+  solc: {
+    version: "/Users/axic/.nvm/versions/node/v8.9.4/lib/node_modules/solc"
+  }
+}
+
+// Optimization and EVM version settings
+compilers: {
+  solc: {
+    settings: {
+      optimizer: {
+        enabled: true, // Default: false
+        runs: 1000     // Default: 200
+      },
+      evmVersion: "homestead"  // Default: "byzantium"
+    }
+  }
+}
+```
+
+### Speed comparison
+Docker and native binary compilers process large contract sets faster than solcjs. If you're just compiling a few contracts at a time, the speedup isn't significant relative to the overhead of running a command (see below). The first time `truffle` uses a docker version there's some extra overhead as it caches the solc version string and a solcjs companion compiler. All subsequent runs should be at full speed.
+
+Times to `truffle compile` on a MacBook Air 1.8GHz, Node 8.11.1
+
+| Project              | # files | solcjs | docker | bin |
+|----------------------|---------:| ------:|--------:|-----------:|
+| truffle/metacoin-box |       3 |   4.4s |   4.4s |      4.7s |
+| gnosis/pm-contracts  |      34 |  21.7s |  10.9s |     10.2s |
+| zeppelin-solidity    |     107 |  36.7s |  11.7s |     11.1s |
+
+
+For help installing a natively built compiler, see the Solidity docs [here](https://solidity.readthedocs.io/en/v0.4.23/installing-solidity.html#binary-packages).
+
+### Contract Profiling
+Truffle has always tried figure out which contracts changed recently and compile the smallest set necessary (with varying degrees of success). It now does this reliably so you may benefit from setting up an `npm` convenience script that looks like this:
+```
+"test": "truffle compile && truffle test"
+```
+
+Many thanks to the Solidity team for making all of the above possible and for their helpful advice over the last year.  🙌 🙌
```

Comment:
I second this ------------------------------------------^

Reply:
That second pair of hands is yours!